src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Completion/LegacyRazorCompletionResolveEndpointTest.cs

```diff
@@ -20,18 +20,18 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion;
 
-public class LegacyRazorCompletionResolveEndpointTest : LanguageServerTestBase
+public class LegacyRazorCompletionResolveEndpointTest(ITestOutputHelper testOutput) : LanguageServerTestBase(testOutput)
 {
-    private readonly LSPTagHelperTooltipFactory _lspTagHelperTooltipFactory;
-    private readonly VSLSPTagHelperTooltipFactory _vsLspTagHelperTooltipFactory;
-    private readonly CompletionListCache _completionListCache;
-    private readonly VSInternalCompletionSetting _completionCapability;
-    private readonly VSInternalClientCapabilities _defaultClientCapability;
-
-    public LegacyRazorCompletionResolveEndpointTest(ITestOutputHelper testOutput)
-        : base(testOutput)
+    private LSPTagHelperTooltipFactory _lspTagHelperTooltipFactory;
+    private VSLSPTagHelperTooltipFactory _vsLspTagHelperTooltipFactory;
+    private CompletionListCache _completionListCache;
+    private VSInternalCompletionSetting _completionCapability;
+    private VSInternalClientCapabilities _defaultClientCapability;
+
+    protected async override Task InitializeAsync()
     {
         var snapshotResolver = new TestSnapshotResolver();
```

Comment:
💡 Maybe make an `Task<ISnapshotResolver> TestSnapshotResolver.CreateAsync` method that initializes as well? Could help clean some of this up and make sure we don't create without initializing

Reply:
This ended up getting cleaned up when I changed `SnapshotResolver` to implement `IOnInitialized`.